test/unit_conv2d.ml

```diff
@@ -0,0 +1,354 @@
+(** Unit test for Convolution3D operations *)
+
+open Owl
+
+module N = Dense.Ndarray.S
+
+(* Functions used in tests *)
+
+let tolerance_f64 = 1e-8
+let tolerance_f32 = 5e-4
+let close a b =
+  N.(a - b |> abs |> sum') < tolerance_f32
+
+let test_conv2d input_shape kernel_shape stride pad =
+  let inp    = N.sequential ~a:1. input_shape  in
+  let kernel = N.sequential ~a:1. kernel_shape in
+  N.conv2d ~padding:pad inp kernel stride
+
+let test_conv2d_back_input input_shape kernel_shape stride pad =
+  let inp     = N.sequential ~a:1. input_shape  in
```

Comment:
unless you can align all the `=` in this function, it doesn't make too much sense to just align some of them, so please make the `=` follow the variable name and leave one whitespace.